src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -106,13 +106,13 @@ class ActionPlanService(
 
   private fun verifySafeForApproval(actionPlan: ActionPlan) {
     if (actionPlan.approvedAt != null) {
-      throw ValidationException("Action plan has already been approved. [id=${actionPlan.id}]")
+      throw ValidationError("Action plan has already been approved", listOf())
```

Comment:
would it not be useful to still include the actionplan id? 
Or are we certain that it's easy to obtain?

Reply:
Ah right, I forgot to mention the reasons behind this. Thanks for reminding me

I was thinking...

- In this particular case, we will know the context of the error is `/action-plan/{id}/approve`, which already gives us enough context
- This error is mapped into a client error, so it'll be coming back on the wire as `{"status": 400, "summary": "validation error", "message": "Action plan has already been approved", "userMessage": null, "validationErrors": [], "accessErrors": null}` which I assumed would be weird if it contained IDs -- would "message" be displayed?

Reply:
@andrewburrowsmoj ^

Reply:
I understand - you're right it makes it more user friendly this way and like you say they can infer it.
I was thinking this was more for our purposes for monitoring rather than api response.